python/cudf/cudf/_lib/csv.pyx

```diff
@@ -318,7 +318,7 @@ def read_csv(
     int skipfooter=0,
     int skiprows=0,
     bool dayfirst=False,
-    str compression="infer",
+    object compression="infer",
```

Comment:
Cython doesn't optimize for `str` in any way so it doesn't matter performance-wise, but why did we need this change? Before, it was clear that Cython was expecting a `str` rather than an arbitrary `object`.

Reply:
A valid input here for `compression` is `None` too. That's why had to change it to make it explicit.